src/cargo/core/resolver/context.rs

```diff
@@ -22,56 +21,26 @@ pub struct ResolverContext {
     pub resolve_features: im_rc::HashMap<PackageId, FeaturesSet, rustc_hash::FxBuildHasher>,
     /// get the package that will be linking to a native library by its links attribute
     pub links: im_rc::HashMap<InternedString, PackageId, rustc_hash::FxBuildHasher>,
-
     /// a way to look up for a package in activations what packages required it
     /// and all of the exact deps that it fulfilled.
     pub parents: Graph<PackageId, im_rc::HashSet<Dependency, rustc_hash::FxBuildHasher>>,
 }
 
+/// By storing activation keys in a `HashMap` we ensure that there is only one
+/// semver compatible version of each crate.
+type Activations = im_rc::HashMap<
+    ActivationKey,
+    (Summary, ContextAge),
+    nohash_hasher::BuildNoHashHasher<ActivationKey>,
```

Comment:
This PR does a number of different things, and I'm trying to understand which ones are critical to the performance win. I tried switching this one `nohash_hasher::BuildNoHashHasher` back to `rustc_hash::FxBuildHasher`, and the Solana only benchmark I was running yesterday runs in 162.53min (or 164.63min there seems to be a surprising amount of run to run variability).

Reply:
So it seems that `rustc-hash` is already very optimized when hashing a single `u64`:
https://github.com/rust-lang/rustc-hash/blob/master/src/lib.rs#L99.

In this case the performance increase may only be due to fact that we don't hash the full `ActivationKeyInner` but only its address.